src/test/java/com/epam/reportportal/cucumber/UtilsTest.java

```diff
@@ -0,0 +1,50 @@
+package com.epam.reportportal.cucumber;
+
+import com.epam.ta.reportportal.ws.model.ParameterResource;
+import gherkin.formatter.Argument;
+import gherkin.formatter.model.Match;
+import org.junit.Test;
+import rp.com.google.common.collect.Lists;
+
+import java.util.Collections;
+import java.util.List;
+
+import static org.junit.Assert.*;
+
+/**
+ * @author <a href="mailto:ihar_kahadouski@epam.com">Ihar Kahadouski</a>
+ */
+public class UtilsTest {
```

Comment:
You verified that your extraction of parameters is correct. But what about sending them to the client? If I delete this line the functionality of parameters will be broken, but your tests will still pass:
https://github.com/reportportal/agent-java-cucumber/blob/7581c29d8159597fa1949cfcc1d81251d4390e45/src/main/java/com/epam/reportportal/cucumber/StepReporter.java#L70

Please add an integration unit-tests.

Reply:
added